Directory.Build.props

```diff
@@ -87,7 +87,7 @@
   <!-- dotnet-gcdump needs net6+ and won't work on mobile. -->
   <PropertyGroup>
     <PlatformIsLegacy Condition="$(TargetFramework.StartsWith('net4')) or $(TargetFramework.StartsWith('netstandard'))">true</PlatformIsLegacy>
-    <PlatformIsMobile Condition="$(TargetFramework.EndsWith('android')) or $(TargetFramework.EndsWith('ios')) or $(TargetFramework.EndsWith('maccatalyst'))">true</PlatformIsMobile>
+    <PlatformIsMobile Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'android' or $([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'ios' or $([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'maccatalyst'">true</PlatformIsMobile>
```

Comment:
See `TargetPlatformIdentifier` defined above

Reply:
Interestingly, I wonder whether by calling it `TargetPlatformIdentifier` we could potentially break things:
- https://github.com/dotnet/msbuild/issues/7359

Some older build scripts might be relying on that property being set to `windows` on older TFMs (`netstandard2.0` and `netstandard2.1` are the only ones that would affect us, I think).

Unrelated to this PR though and we haven't run into any issues yet (that I know of)... but we could rename it just to avoid any potential clash.